ludwig/utils/schema_utils.py

```diff
@@ -116,6 +122,43 @@ def _jsonschema_type_mapping(self):
             ]
         }
 
+class InitializerOptionsOrCustomDictField(fields.Field):
+    def _deserialize(self, value, attr, data, **kwargs):
+        # if value is None:
+        #     return value
+
+        if isinstance(value, str):
+            initializers = list(initializer_registry.keys())
+            if value not in initializers:
+                raise ValidationError(
+                    f"Expected one of: {initializers}, found: {value}"
+                )
+            return value
+
+        if isinstance(value, dict):
+            if "type" not in value:
+                raise ValidationError(
+                    f"Dict must contain 'type'"
+                )
+            return value
+
+        raise ValidationError('Field should be str or dict')
+
+    def _jsonschema_type_mapping(self):
+        return {
+            'oneOf': [
+                {'type': 'string', 'enum': list(initializer_registry.keys())},
+                {
+                    "type": "object",
+                    "properties": {
+                        "type": { "type": "string" },
```

Comment:
Should type also be one of the list of `initializer_registry`?

Reply:
True!